### DISCUSSION TITLE
See which provider was used to login with OAuth2

### DISCUSSION THREAD

**Discussion by Fish1** [pocketbase/pocketbase#2161] [Q&A] [answered]
Is it possible to see which provider was used to login with OAuth2? I am looking to do some deeper discord integration with my application. But how would I verify, on the server that it was infact discord a user logged in with, and not google?

I noticed this post where we now have access to the "rawUser". But how would I know, before I access the raw user, if the data is for discord or a different OAuth2 provider?

https://github.com/pocketbase/pocketbase/discussions/654#discussioncomment-4273677

**Comment by ganigeorgiev** [comment, marked as answer]
> But how would I verify, on the server that it was infact discord a user logged in with, and not google?

Hm, there is no easy way to do this at the moment. I'll consider exposing the initialized provider, or at least its name, as part of the `RecordAuthWithOAuth2Event` struct in the next release.

As a temporary workaround for now you can check the the `rawData` for a field that is unique to discord (eg. `discriminator`) and/or check the `_externalAuths` table for a linked entry with the provider id (_although there is no guarantee for id uniqueness between the providers_):

```go
app.OnRecordAfterAuthWithOAuth2Request().Add(func(e *core.RecordAuthWithOAuth2Event) error {
	externalAuth, _ := app.Dao().FindExternalAuthByRecordAndProvider(e.Record, "discord")

	isDiscord := externalAuth != nil && externalAuth.ProviderId == e.OAuth2User.Id

	log.Println(isDiscord)

	return nil
})
```

**Reply by Fish1** [reply]
Thanks for the quick response! I'll try that out.
